syntax = "proto3";

package calculator;

// The request message containing two double values.
message MultRequest {
  Fraction frac1 = 1;
  Fraction frac2 = 2;
}

message Fraction{
  int32 num = 1;
  int32 den = 2;
}

// The response message containing the result of the sum.
message MultResponse {
  Fraction result = 1;
}

// The calculator service definition.
service Calculator {
  // Sends a SumRequest and receives a SumResponse.
  rpc Mult (MultRequest) returns (MultResponse);
}
